extensions/infinispan-embedded/runtime/pom.xml

```diff
@@ -30,7 +30,6 @@
             <groupId>io.smallrye</groupId>
             <artifactId>smallrye-metrics</artifactId>
         </dependency>
```

Comment:
You should remove that one and use `quarkus-smallrye-metrics` here and `quarkus-smallrye-metrics-deployment` in the deployment module.

Reply:
Sounds good. I have no clue about this metrics stuff :)

Reply:
I think there is no deployment-time dependency on metrics in this case. Metrics are handled by Infinispan directly, not by any Quarkus extension, so it all happens in runtime

Reply:
So I must admit I am not sure what you guys want me to do :)

Reply:
@jmartisk we might need some reflection registration at some point and so on.

So if you have the runtime dependency, you need the deployment one.

Reply:
I pushed an additional commit with the change.

@jmartisk can you check it's OK with you?

Reply:
@gsmet sorry I could have done that - I was more asking if I needed to actually do that.

Reply:
Ah OK, I misunderstood. Let's have @jmartisk check this. I think it's the right thing to do to be on the safe side and be sure SmallRye Metrics behave correctly in native mode. Even if the deployment extension doesn't have anything special for native atm (I haven't checked), it might not always be the case so better be on the safe side.

Reply:
LGTM!